src/node/internal/internal_timers.ts

```diff
@@ -54,18 +54,22 @@ export class Timeout {
   }
 
   #constructTimer(): number {
-    // @ts-expect-error TS2322 Due to difference between Node.js and globals
+    // Use raw timer primitives from TimersUtil to avoid circular dependency.
```

Comment:
As an alternative to defining the raw versions on timers.h, you could capture the original impls and defer to them...

```js
const origSetInterval = globalThis.setInterval.bind(globalThis);
const origSetTimeout = globalThis.setTimeout;.bind(globalThis);

#constructTimer(): number {
  this.#timer = this.#isRepeat
     ? origSetInterval(this.#callback, this.#after, ...this.#args)
     : origSetTimeout(this.#callback, this.#after, ...this.#args);
}

// ...

globalThis.setInterval = function setInterval ....
globalThis.setTimeout = function setTimeout ....
```

This would likely be an easier way to avoid the circ dependency.

Reply:
By the time node timers internal is evaluated the global scope was already overridden

Remember that a user can call setTimeout *before* importing node:timers